### PR TITLE
chore(core): eslint rule to avoid relative imports

### PR DESCRIPTION
**Pull request by alemagio**
## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/server/eslint.config.mjs [hunk 1]
```diff
@@ -38,6 +38,12 @@ const configs = [
       }
     },
     rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: ['.*']
```
Comment (fabis94) [inline packages/server/eslint.config.mjs:44]: How's this work? Doesn't a `.*` regex resolve to literally any kind of string?

**Review by fabis94 — CHANGES_REQUESTED** [review]

File: packages/server/eslint.config.mjs (hunk 1 above)
Comment (fabis94) [reply packages/server/eslint.config.mjs:44]: Ah, these can be glob patterns too apparently. Well, in any case, there's some paths that need fixing picked up by pre-commit

Comment (fabis94) [reply packages/server/eslint.config.mjs:44]: @iainsproat That's alright, we can add an eslint-disable comment there, cause that's an exception scenario.

File: packages/server/eslint.config.mjs [hunk 2]
```diff
@@ -38,6 +38,12 @@ const configs = [
       }
     },
     rules: {
+      'no-restricted-imports': [
```
Comment (fabis94) [inline packages/server/eslint.config.mjs:41]: Why is this only assigned to '*.ts' files?

Comment (alemagio) [reply packages/server/eslint.config.mjs:41]: I assume at some point we will move all the files to .ts right?
Do we want to enforce the rule also in js files?

**Review by fabis94 — APPROVED** [review]

**Pull request merged**